content/articles/building-a-simple-translation-app-using-python-for-beginners/index.md

```diff
@@ -0,0 +1,171 @@
+
+![Installing PyTorch](/engineering-education/building-a-simple-translation-app-using-python-for-beginners/installing-pytorch.png)
+
+*[Image Source: PyTorch](https://pytorch.org/)*
+
+The commands for installing PyTorch vary depending on factors such as your OS, package of choice (pip, conda, source), language, and compute platform (CUDA, CPU, or ROCm). So, make sure you select your preference. It doesn't have to look like mine. Mine's shown below:
+
+![Preference](/engineering-education/building-a-simple-translation-app-using-python-for-beginners/pytorch-preference.png)
+
+*[Image Source: PyTorch](https://pytorch.org/)*
+
+To install PyTorch on my machine, I've selected the Long Term Support (LTS 1.8.1) build, Linux OS, `pip` package, Python programming language, and CPU as my compute platform of choice. If your machine has GPU support, choose accordingly.  
+
+After selecting those options, this is the command you're going to run to install PyTorch:
+
+```python
+    pip3 install torch==1.8.1+cpu torchvision==0.9.1+cpu torchaudio==0.8.1 -f https://download.pytorch.org/whl/lts/1.8/torch_lts.html
+```
+Let's now go ahead and install our other dependencies, the Hugging Face Transformers and Gradio. 
+```python
+    pip install transformers ipywidgets gradio --upgrade
+```
+> Please note that this command can be used in the command-line interface for a local machine and can also be run as a cell in the Colab notebook. You only add an exclamation `!` mark before the `pip` command when installing on the Colab. This way: `!pip install transformers ipywidgets gradio --upgrade`
+
+Our dependencies are now installed successfully!
+
+`transformers` is going to give us our translation pipeline.
+`ipywidgets` provides you with the progress bar as you're downloading the model.
+`gradio` gives you a nice way to demonstrate and interact with your ML model.
+
+Now that we've installed our dependencies, the next step involves importing them into our model.
+
+To import them, let's issue the following commands:
+
+```python
+    import gradio as gr
+    from transformers import pipeline
+```
+Now that that is done, let's go ahead and download our pre-trained translation model.
+
+### Downloading the pre-trained translation model
+Hugging Face has many Transformer pipelines. You can check them on this [documentation](https://huggingface.co/transformers/main_classes/pipelines.html). For our tutorial, we will select the `TranslationPipeline` which is a task-specific pipeline that allows the translation from one language to another. In the documentation, they state that you can use the `pipeline()` method and the task identifier `translation_xx_to_yy` to translate. `xx` being the language you want to translate from and `yy` the language you want to translate to. For our experiment, we are building a simple translation app that can translate from English to French. 
```

Comment:
1. The "T" in "Transformer" need not be capitalized. 
2. Add a comma after `TranslationPipeline`.
3. Remove the word "the" in the phrase "that allows the translation from one language to another"
4. The fullstop must be replaced with a comma in the phrase "to translate. `xx` being the language..."
5. Please replace the word "experiment" with an appropriate word.